pyvera/__init__.py

```diff
@@ -466,8 +466,13 @@ def get_value(self, name):
 
         This data is updated by the subscription service.
         """
+        return self.get_strict_value(name.lower())
+
+    def get_strict_value(self, name):
```

Comment:
Can see why this is needed. I added the `lower` because the keys from `get_complex_values` are camel case - but the ones in deviceInfo are _mostly_ lower case.

Whatever we do at this point is ugly - but I wonder if it's better to keep to a single `get_value` call for now - and have it attempt both the passed key - and a lower case value. So something like 

```
    def get_value(self, name):
        """Get a value from the dev_info area.

        This is the common Vera data and is the best place to get state from
        if it has the data you require.

        This data is updated by the subscription service.
        """
        dev_info = self.json_state.get('deviceInfo')
        ret = dev_info.get(name.lower(), None)
        return ret if ret is not None else dev_info.get(name, None)
```

What do you think?

Reply:
That would be fine as well.  Let me know if you’d like me to make that change.

Reply:
I think this should be formalized as separate, not homogenized, and here's what I mean.

The service state values, which come from the status request to Vera, are the master set of device values and states. These are the Pascal-case names.

The shortcode values, which come from the sdata request to Vera, are a small subset of the service state that is defined by each device's service file (S_device.XML). 

I think get_complex_value() should be deprecated and a new method added named get_service_state() or get_service_value(), which is more in keeping with Vera's/UPnP's terminology and familiar to the Vera developer community, and get_value() should be deprecated and a new method added called get_shortcode_value() or similar. 

Vera clients should be well aware of the differences between these values. Combining them into a single get_value() call creates the opportunity for collisions, and more confusion than this odd disconnect by Vera already creates. 

It's worth nothing that the status request (full state variable results) can also accept the DataVersion parameter, to return incremental changes to prior requests, as lu_sdata does in get_changed_devices(). In my perfect world, pyvera would not have used the shortcode/sdata here, but rather status, thus making the full suite of changes a device could see available to subscribers. As it stands today, only the subset is published.

Reply:
Some really good points toggledbits, I would also prefer the full suite for status returned, maybe this is worth documenting as a feature request?

If the end goal is to make these non-homogeneous, then maybe it would be worth renaming these functions now as you describe and then we make another pull request to split them as you described?

Reply:
I would not rename them, as that would break existing clients. When we get around to implementing the new names, just comment the old methods "deprecated" (leave them in place and functional) and give the world a long while to change to the new methods before removing the old.

Reply:
Ok - left as-is for now, can work it later